### DISCUSSION TITLE
Trouble configuring SSL connection with Azure MySQL flexible server

### DISCUSSION THREAD

**Discussion by daniel-hampton** [prisma/prisma#19931] [Q&A]
I am having trouble getting Prisma to connect to an Azure MySQL Flexible Server instance with SSL.

I can connect using the MySQL CLI. 
This works.
```bash
mysql -h my-db-name.database.azure.com -u azuredbadmin -p --ssl-mode=REQUIRED
```

I can also download the SSL public certificate (`DigiCertGlobalRootCA.crt.pem`) from the Azure dashboard:
![image](https://github.com/prisma/prisma/assets/20423706/9aef439a-390e-4194-93ab-dc8b49a4fdc6)

Connecting with Prisma via DATABASE_URL does not work:
```bash
DATABASE_URL="mysql://myusername:mypassword@my-db-name.database.azure.com:3306/payments?sslcert=./DigiCertGlobalRootCA.crt.pem"
```
I put the file inside the prisma folder
![image](https://github.com/prisma/prisma/assets/20423706/abfae449-85c9-4870-9a88-ccae8010624c)

Command and received error:
```bash
dotenv -e .env.local npx prisma migrate status
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "payments" at "my-db-name.mysql.database.azure.com:3306"

Error: P1000: Authentication failed against database server at `my-db-name.mysql.database.azure.com`, the provided database credentials for `myusername` are not valid.

Please make sure to provide valid database credentials for the database server at `my-db-name.mysql.database.azure.com`.
```

The suggested url in the [prisma docs](https://www.prisma.io/docs/concepts/database-connectors/mysql#configuring-an-ssl-connection) has a few extra parameters `sslidentity` and `sslpassword`. I'm unsure how to generate these if I only have access to the public certificate. I'm also unsure why I don't see them required in any of the other connection examples given in the Azure console's MySQL connect page.

```bash
mysql://USER:PASSWORD@HOST:PORT/DATABASE?sslidentity=client-identity.p12&sslpassword=mypassword&sslcert=rootca.cert
```

Azure's app connection suggestions: 
![image](https://github.com/prisma/prisma/assets/20423706/f394ffec-fd73-4233-884f-9203f48aabb3)

Any help is appreciated. Thank you.

**Comment by daniel-hampton** [comment]
I further tried the [steps here](https://github.com/prisma/prisma/issues/2676) to generate a client key, identity file, and self signed certificate. This process seems strange to me since Azure doesn't seem to want these. I get a different error, even with `sslaccept=accept_invalid_certs`

```bash
DATABASE_URL="mysql://USERNAME:PASSWORD@my-db-name.mysql.database.azure.com:3306/payments?sslaccept=accept_invalid_certs&sslcert=DigiCertGlobalRootCA.crt.pem&sslidentity=client-identity.p12&sslpassword=certpassword"
```

```bash
Error: P1011: Error opening a TLS connection: error:0A000418:SSL routines:ssl3_read_bytes:tlsv1 alert unknown ca:../ssl/record/rec_layer_s3.c:1584:SSL alert number 48
```

**Reply by nurul3101** [reply]
Hello @daniel-hampton 👋 

Did you try using an absolute path instead of a relative path while passing the value of `sslcert`?
```
DATABASE_URL="mysql://myusername:mypassword@my-db-name.database.azure.com:3306/payments?sslcert=./DigiCertGlobalRootCA.crt.pem"
```

**Reply by daniel-hampton** [reply]
That example still is a relative path, the `./filename`

I believe I did it like that early on, I'll try again but it didn't match the examples I was seeing.

**Reply by nurul3101** [reply]
@daniel-hampton Did you manage to solve it?

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
Please feel free to let us know in case you are still facing this issue.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
